query-node/joystream-query-node/README.md

```diff
@@ -24,6 +24,81 @@ $ cd generated/indexer
 $ yarn start
 ```
 
+## Add a new mapping for Joystream MemberRegistered event
+
+1. Every mapping function get a parameter of `QueryEvent` type
+
+```ts
+import { QueryEvent, db } from "../generated/indexer/index";
+```
+
+2. `db` object for database operations: save/get/remove
+
+```ts
+import { db } from "../generated/indexer/index";
+```
+
+3. Define the event handler function with the following signature
+
+```ts
+export async function handleMemberRegistered(event: QueryEvent) {}
+```
+
+4. Inside the handler function create a new instance of the entity and fill properties with event data.
+
+```ts
+const member = new MemberRegistereds();
+// Get event data
+const { AccountId, MemberId } = event.event_params;
+
+member.accountId = AccountId;
+member.memberId = +MemberId;
+```
+
+5. Call `db.save()` method to save data on database
+
+```ts
+// Save to database.
+db.save(member, event);
+```
+
+6. Query database
+
+```ts
+// Query from database
+const m = await db.get("MemberRegistereds", { memberId: MemberId });
+console.log(m);
+```
+
+**Notes**
+`db.save()` method take two parameters entity instance and event. `db.save()` method use `event` to keep track of last processed event.
+
+`db.get()` method. The first parameter is the name of the entity class to query and the second parameter is the find options (to match record).
+
+**Complete code**
+
+```ts
+import { MemberRegistereds } from "../generated/indexer/entities/MemberRegistereds";
+import { QueryEvent, db } from "../generated/indexer/index";
+
+export async function handleMemberRegistered(event: QueryEvent) {
+  const member = new MemberRegistereds();
```

Comment:
Does the `MemberRegistereds` not have a proper constructor?

Reply:
No, it doesn't have. But I can add it. By adding an option to `codegen` command.